src/swsssdk/dbconnector.py

```diff
@@ -9,182 +9,324 @@
+        # Check if the namespace, db_name is valid
+        if self.check_instance_valid(namespace) == False or db_name not in self.get_db_list(namespace):
+            message = "Namespace or Database name provided is not valid!"
+            logger.exception(message)
+            raise RuntimeError(message)
+
+        #set a database name as a constant value attribute.
+        for db_name in self.get_db_list(namespace):
             setattr(self, db_name, db_name)
 
-    def connect(self, db_name, retry_on=True):
         if self.use_unix_socket_path:
-            self.redis_kwargs["unix_socket_path"] = self.get_db_socket(db_name)
-            self.redis_kwargs["host"] = None
-            self.redis_kwargs["port"] = None
+            self.db_intfs[namespace].redis_kwargs["unix_socket_path"] = self.get_db_socket(db_name, namespace)
+            self.db_intfs[namespace].redis_kwargs["host"] = None
+            self.db_intfs[namespace].redis_kwargs["port"] = None
         else:
-            self.redis_kwargs["host"] = self.get_db_hostname(db_name)
-            self.redis_kwargs["port"] = self.get_db_port(db_name)
-            self.redis_kwargs["unix_socket_path"] = None
-        db_id = self.get_dbid(db_name)
-        super(SonicV2Connector, self).connect(db_id, retry_on)
+            self.db_intfs[namespace].redis_kwargs["host"] = self.get_db_hostname(db_name, namespace)
+            self.db_intfs[namespace].redis_kwargs["port"] = self.get_db_port(db_name, namespace)
+            self.db_intfs[namespace].redis_kwargs["unix_socket_path"] = None
+
+        db_id = self.get_dbid(db_name, namespace)
+        self.db_intfs[namespace].connect(db_id, retry_on)
+        self.clients[namespace] = self.db_intfs[namespace].get_redis_client(db_id, namespace)
+
+    """
+    Connect to the DB instance in the "namespaces" specified. It is a list.
+	* [] : It connects to the local 'db_name' instance.
+        * list of namespaces (1 or more ) to connect to [asic0, asic1, asic2 ...]
+    """
+    def connect(self, db_name, retry_on=True, namespaces=[]):
+        if namespaces:
+            ns_instances = namespaces
+
+            # The TCP connection to a DB in different namespace in not supported.
+            if self.use_unix_socket_path == False:
+                message = "TCP connectivity to the DB instance in a different namespace is not implemented!"
+                logger.exception(message)
+                raise NotImplementedError(message)
+        else
+            ns_instances = ['default']
+
+        for instance in ns_instances:
+            self.db_connect(db_name, retry_on, instance)
 
     def close(self, db_name):
-        db_id = self.get_dbid(db_name)
-        super(SonicV2Connector, self).close(db_id)
+        for instance in ns_instances:
+            db_id = self.get_dbid(db_name, instance)
+            self.db_intfs[instance].close(db_id)
+
+    def check_instance_valid(self, namespace='default'):
```

Comment:
>check_instance_valid [](start = 8, length = 20)

Why this is needed? Give a sample use case?

Reply:
The idea was to call this API, when the user passes the namespace as an argument. We take the argument he passed, and check if it is there in the list of namespaces which we get by parsing the database_config.json files. 

I have updated the usage now with a similar API namespace_validation() introduced to check the namespace validation similar to instance/db_name. This is better than doing this check at the connect() APIs.